### PR TITLE
PoC of extensibility mechanism

### PR DESCRIPTION
**Pull request by symphony-elias**
### Ticket
#479 

### Description
PoC demonstrating what could be an extensibility mechanism in the BDK (core and Spring Boot starter).

### CODE REVIEW

File: symphony-bdk-examples/bdk-spring-boot-example/src/main/java/com/symphony/bdk/examples/spring/EchoActivity.java [hunk 1]
```diff
@@ -21,6 +22,9 @@
   @Autowired
   private MessageService messageService;
 
+  @Autowired
+  private MyExtension extension;
```
Comment (symphony-youri) [inline symphony-bdk-examples/bdk-spring-boot-example/src/main/java/com/symphony/bdk/examples/spring/EchoActivity.java:26]: Cool

File: symphony-bdk-core/src/main/java/com/symphony/bdk/core/SymphonyBdk.java [hunk 2]
```diff
@@ -42,6 +49,7 @@
 public class SymphonyBdk {
 
   private final BdkConfig config;
+  private final List<Extension> extensions;
```
Comment (symphony-youri) [inline symphony-bdk-core/src/main/java/com/symphony/bdk/core/SymphonyBdk.java:52]: what is the value of storing over instanciating them on the fly in the getExtension() method? 
I would say to keep the same lifecycle as other services (but just to make sure and to open the discussion)

**Review by symphony-youri — COMMENTED** [review]
Maybe we should name them ServiceExtension? (a bit long...) to avoid blocking the concept of extension in case we come up with other extension points?

File: symphony-bdk-core/src/main/java/com/symphony/bdk/core/SymphonyBdk.java (hunk 2 above)
Comment (symphony-elias) [reply symphony-bdk-core/src/main/java/com/symphony/bdk/core/SymphonyBdk.java:52]: Indeed, the added value is low and we could instantiate them on the fly.

**Pull request closed without merging**